src/tests/src/libs/antares/study/constraint/test_constraint.cpp

```diff
@@ -15,6 +15,15 @@
 using namespace Antares::Data;
 namespace fs = std::filesystem;
 
+fs::path generateAndCreateDirName()
+{
+    std::string dir = boost::unit_test::framework::current_test_case().p_name;
+    fs::path working_dir = fs::temp_directory_path() / dir;
+    fs::remove(working_dir);
+    fs::create_directories(working_dir);
+    return working_dir;
+}
+
 BOOST_AUTO_TEST_SUITE(BindingConstraintTests)
```

Comment:
You may need to remove unwanted characters, not all characters are valid in a path

Also, please use a single declaration/definition

Reply:
I don't see which unwanted characters could be found, we're only using alpha numeric chars and underscores in test names.

This function is used in different places, should it be moved in a header at the root of src/tests ?